source/stylist/fortran.py

```diff
@@ -295,3 +297,58 @@ def examine_fortran(self, subject: FortranSource) -> List[Issue]:
 
         issues.sort(key=lambda x: (x.filename, x.line, x.description))
         return issues
+
+
+class KindPattern(FortranRule):
+    def __init__(self,
+                 integer_pattern: Union[str, Pattern],
+                 real_pattern: Union[str, Pattern]):
+        self._patterns: Dict[str, Pattern] \
+            = defaultdict(lambda: re.compile(r'.*'))
+        if isinstance(integer_pattern, str):
+            self._patterns['integer'] = re.compile(integer_pattern)
+        else:
+            self._patterns['integer'] = integer_pattern
+        if isinstance(real_pattern, str):
+            self._patterns['real'] = re.compile(real_pattern)
+        else:
+            self._patterns['real'] = real_pattern
+
+    def examine_fortran(self, subject: FortranSource) -> List[Issue]:
+        issues: List[Issue] = []
+
+        candidates: List[Fortran2003.Base] = []
+        # Component variables
+        candidates.extend(
+            subject.find_all(Fortran2003.Data_Component_Def_Stmt))
+        # Component Procedure
+        candidates.extend(
+            subject.find_all(Fortran2003.Proc_Component_Def_Stmt))
+        # Procedure declaration
+        candidates.extend(
+            subject.find_all(Fortran2003.Procedure_Declaration_Stmt))
+        # Variable declaration
+        candidates.extend(
+            subject.find_all(Fortran2003.Type_Declaration_Stmt))
+
+        for candidate in candidates:
+            if isinstance(candidate,  # Is variable
+                          (Fortran2003.Data_Component_Def_Stmt,
+                           Fortran2003.Type_Declaration_Stmt)):
+                type_spec: Fortran2003.Intrinsic_Type_Spec = candidate.items[0]
+                kind_selector: Fortran2003.Kind_Selector = type_spec.items[1]
+
+                if isinstance(kind_selector, Fortran2003.Kind_Selector):
+                    data_type: str = type_spec.items[0].lower()
+                    kind: str = kind_selector.string[1:-1]
```

Comment:
Is the `[1:-1]` here intended for stripping off quotation marks?  Could use `.strip` instead if so...

Reply:
It is for removing the brackets but you are basically right. I was not aware of the `.strip()` functionality so have changed it as it is much more readable.